File: tool/generate_icon.dart
// Generates app icon as SVG, then we convert to PNG
// Run: dart run tool/generate_icon.dart

import 'dart:io';

void main() {
  const size = 1024;
  const half = size / 2;

  // Math + AI Bot icon: A friendly robot face with math symbols
  // on a vibrant blue-purple gradient background
  final svg = '''<?xml version="1.0" encoding="UTF-8"?>
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 $size $size" width="$size" height="$size">
  <defs>
    <linearGradient id="bg" x1="0%" y1="0%" x2="100%" y2="100%">
      <stop offset="0%" style="stop-color:#4F46E5"/>
      <stop offset="100%" style="stop-color:#7C3AED"/>
    </linearGradient>
    <linearGradient id="botBody" x1="0%" y1="0%" x2="0%" y2="100%">
      <stop offset="0%" style="stop-color:#FFFFFF"/>
      <stop offset="100%" style="stop-color:#E0E7FF"/>
    </linearGradient>
  </defs>

  <!-- Background with rounded corners -->
  <rect width="$size" height="$size" rx="220" fill="url(#bg)"/>

  <!-- Math symbols scattered in background -->
  <text x="140" y="220" font-family="Arial" font-size="120" font-weight="bold" fill="rgba(255,255,255,0.15)" transform="rotate(-15,140,200)">+</text>
  <text x="780" y="250" font-family="Arial" font-size="100" font-weight="bold" fill="rgba(255,255,255,0.15)" transform="rotate(10,800,230)">×</text>
  <text x="100" y="850" font-family="Arial" font-size="110" font-weight="bold" fill="rgba(255,255,255,0.15)" transform="rotate(20,120,830)">÷</text>
  <text x="800" y="880" font-family="Arial" font-size="90" font-weight="bold" fill="rgba(255,255,255,0.12)" transform="rotate(-10,820,860)">π</text>
  <text x="150" y="530" font-family="Arial" font-size="80" font-weight="bold" fill="rgba(255,255,255,0.1)">∑</text>
  <text x="820" y="580" font-family="Arial" font-size="85" font-weight="bold" fill="rgba(255,255,255,0.1)">%</text>

  <!-- Robot head (rounded rectangle) -->
  <rect x="270" y="280" width="484" height="420" rx="80" fill="url(#botBody)" stroke="#C7D2FE" stroke-width="8"/>

  <!-- Antenna -->
  <line x1="$half" y1="280" x2="$half" y2="200" stroke="#C7D2FE" stroke-width="12" stroke-linecap="round"/>
  <circle cx="$half" cy="185" r="25" fill="#34D399"/>
  <circle cx="$half" cy="185" r="12" fill="#6EE7B7"/>

  <!-- Eyes -->
  <circle cx="400" cy="460" r="55" fill="#4F46E5"/>
  <circle cx="624" cy="460" r="55" fill="#4F46E5"/>
  <!-- Eye highlights -->
  <circle cx="418" cy="445" r="18" fill="#FFFFFF"/>
  <circle cx="642" cy="445" r="18" fill="#FFFFFF"/>
  <!-- Eye inner -->
  <circle cx="400" cy="460" r="20" fill="#1E1B4B"/>
  <circle cx="624" cy="460" r="20" fill="#1E1B4B"/>

  <!-- Smile -->
  <path d="M 420 560 Q $half 630 604 560" fill="none" stroke="#4F46E5" stroke-width="14" stroke-linecap="round"/>

  <!-- Ears / side panels -->
  <rect x="228" y="410" width="50" height="100" rx="20" fill="#C7D2FE"/>
  <rect x="746" y="410" width="50" height="100" rx="20" fill="#C7D2FE"/>

  <!-- Math on the forehead: "1+1" -->
  <text x="$half" y="385" font-family="Arial" font-size="64" font-weight="bold" fill="#4F46E5" text-anchor="middle">1+1</text>

  <!-- Bottom text: "AI" badge -->
  <rect x="430" y="740" width="164" height="64" rx="32" fill="#34D399"/>
  <text x="$half" y="784" font-family="Arial" font-size="40" font-weight="bold" fill="#FFFFFF" text-anchor="middle">AI</text>
</svg>''';

  final file = File('assets/icon/app_icon.svg');
  file.writeAsStringSync(svg);
  print('SVG icon written to assets/icon/app_icon.svg');
  print('Convert to PNG using: magick convert assets/icon/app_icon.svg -resize 1024x1024 assets/icon/app_icon.png');
}

File: tool/generate_icon.py
"""Generate AIMathTest app icon using Pillow (no Cairo needed)."""
from PIL import Image, ImageDraw, ImageFont
import os

SIZE = 1024
HALF = SIZE // 2

img = Image.new('RGBA', (SIZE, SIZE), (0, 0, 0, 0))
draw = ImageDraw.Draw(img)

# Background: rounded rectangle with gradient effect
# Since Pillow doesn't do gradients natively, we'll simulate with bands
for y in range(SIZE):
    r = int(79 + (124 - 79) * y / SIZE)   # 4F -> 7C
    g = int(70 + (58 - 70) * y / SIZE)    # 46 -> 3A
    b = int(229 + (237 - 229) * y / SIZE)  # E5 -> ED
    draw.line([(0, y), (SIZE, y)], fill=(r, g, b, 255))

# Apply rounded corner mask
mask = Image.new('L', (SIZE, SIZE), 0)
mask_draw = ImageDraw.Draw(mask)
mask_draw.rounded_rectangle([0, 0, SIZE, SIZE], radius=220, fill=255)
img.putalpha(mask)

# Try to load a nice font, fall back to default
try:
    font_large = ImageFont.truetype("arial.ttf", 120)
    font_med = ImageFont.truetype("arialbd.ttf", 64)
    font_small = ImageFont.truetype("arialbd.ttf", 40)
    font_symbols = ImageFont.truetype("arial.ttf", 100)
    font_eyes = ImageFont.truetype("arialbd.ttf", 48)
except:
    font_large = ImageFont.load_default()
    font_med = font_large
    font_small = font_large
    font_symbols = font_large
    font_eyes = font_large

# Background math symbols (subtle)
sym_color = (255, 255, 255, 35)
draw.text((120, 130), "+", fill=sym_color, font=font_symbols)
draw.text((780, 160), "×", fill=sym_color, font=font_symbols)
draw.text((90, 760), "÷", fill=sym_color, font=font_symbols)
draw.text((800, 790), "π", fill=sym_color, font=font_symbols)
draw.text((140, 460), "∑", fill=(255, 255, 255, 25), font=font_symbols)
draw.text((810, 490), "%", fill=(255, 255, 255, 25), font=font_symbols)

# Robot head (rounded rectangle)
head_color = (255, 255, 255, 255)
head_border = (199, 210, 254, 255)
draw.rounded_rectangle([270, 280, 754, 700], radius=80, fill=head_color, outline=head_border, width=8)

# Antenna line
draw.line([(HALF, 280), (HALF, 200)], fill=head_border, width=12)
# Antenna ball
draw.ellipse([HALF - 25, 160, HALF + 25, 210], fill=(52, 211, 153, 255))
draw.ellipse([HALF - 12, 173, HALF + 12, 197], fill=(110, 231, 183, 255))

# Eyes
eye_color = (79, 70, 229, 255)  # #4F46E5
pupil_color = (30, 27, 75, 255)  # #1E1B4B
white = (255, 255, 255, 255)

# Left eye
draw.ellipse([345, 405, 455, 515], fill=eye_color)
draw.ellipse([378, 440, 414, 476], fill=white)  # highlight
draw.ellipse([380, 440, 420, 480], fill=pupil_color)

# Right eye
draw.ellipse([569, 405, 679, 515], fill=eye_color)
draw.ellipse([602, 440, 638, 476], fill=white)  # highlight
draw.ellipse([604, 440, 644, 480], fill=pupil_color)

# Smile (arc)
draw.arc([400, 510, 624, 640], start=10, end=170, fill=eye_color, width=14)

# Ear panels
draw.rounded_rectangle([228, 410, 278, 510], radius=20, fill=head_border)
draw.rounded_rectangle([746, 410, 796, 510], radius=20, fill=head_border)

# Math on forehead: "1+1"
bbox = draw.textbbox((0, 0), "1+1", font=font_med)
tw = bbox[2] - bbox[0]
draw.text((HALF - tw // 2, 310), "1+1", fill=eye_color, font=font_med)

# "AI" badge at bottom
badge_color = (52, 211, 153, 255)  # #34D399
draw.rounded_rectangle([430, 740, 594, 804], radius=32, fill=badge_color)
bbox = draw.textbbox((0, 0), "AI", font=font_small)
tw = bbox[2] - bbox[0]
draw.text((HALF - tw // 2, 752), "AI", fill=white, font=font_small)

# Save
out_path = os.path.join('assets', 'icon', 'app_icon.png')
# Flatten to RGB with background color for non-transparent version
bg = Image.new('RGBA', (SIZE, SIZE), (79, 70, 229, 255))
bg.paste(img, mask=img)
bg.save(out_path, 'PNG')
print(f'Generated {out_path} ({SIZE}x{SIZE})')

# Also generate adaptive icon foreground (with padding for Android adaptive)
adaptive = Image.new('RGBA', (SIZE, SIZE), (0, 0, 0, 0))
adaptive.paste(img, mask=img)
adaptive_path = os.path.join('assets', 'icon', 'app_icon_adaptive.png')
adaptive.save(adaptive_path, 'PNG')
print(f'Generated {adaptive_path} ({SIZE}x{SIZE})')
